### ISSUE TITLE
setting MPLCONFIGDIR in  __init__ as unicode breaks env on Python 2.7 for Windows

### ISSUE THREAD

**Issue by mindw** [matplotlib/matplotlib#5666]
If _create_tmp_config_dir() is called for whatever reason on Python 2.7 for windows, It sets MPLCONFIGDIR as unicode (due to the use of unicode_literals).
This makes the environment broken for usage in subprocess calls as unicode environment variable names are prohibited on this configuration.
This crops up for example when building wcsaxis with setuptools_scm installed:

``` python
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\setuptools_scm\integration.py", line 31, in find_files
    return do(ep.load(), path).splitlines()
  File "C:\Python27\lib\site-packages\setuptools_scm\utils.py", line 74, in do
    out, err, ret = do_ex(cmd, cwd)
  File "C:\Python27\lib\site-packages\setuptools_scm\utils.py", line 60, in do_ex
    env=env
  File "C:\Python27\lib\subprocess.py", line 710, in __init__
    errread, errwrite)
  File "C:\Python27\lib\subprocess.py", line 958, in _execute_child
    startupinfo)
```

**Comment by tacaswell** [comment]
What version of mpl are you using?

**Comment by mindw** [comment]
1.5.0.

**Comment by tacaswell** [comment]
Is this a regression form 1.4.3?  It looks like the relevant section of code has been there from 2012 ( 43ae77e4a4c8bc801c6238b8315de336298f7830 )

**Comment by tacaswell** [comment]
might have also been broken by f4adec7b569cfd0b30e0f8367ba8618b9e160f92 which added unicode literals.

**Comment by embray** [comment]
There is  related issue also broken by the use of `unicode_literals` in `matplotlib.__init__`: https://trac.sagemath.org/ticket/24379

I would suggest against `unicode_literals`.  Even Guido thinks it was a mistake.

**Comment by anntzer** [comment]
@embray I think this can be closed based on your recent PR?

**Comment by embray** [comment]
Quite likely, though it might be good to add a test.

**Comment by anntzer** [comment]
Closing as "likely fixed" + "2.2 is a maintenance-only branch" + "3.0 won't support Py2".
